SDWebImage/SDImageCache.m

```diff
@@ -217,31 +230,34 @@ - (void)storeImage:(nullable UIImage *)image
     
     if (toDisk) {
         dispatch_async(self.ioQueue, ^{
+            NSError * writeError = nil;
             @autoreleasepool {
                 NSData *data = imageData;
                 if (!data && image) {
                     SDImageFormat imageFormatFromData = [NSData sd_imageFormatForImageData:data];
                     data = [image sd_imageDataAsFormat:imageFormatFromData];
                 }                
-                [self storeImageDataToDisk:data forKey:key];
+                [self storeImageDataToDisk:data forKey:key error:&writeError];
             }
             
             if (completionBlock) {
                 dispatch_async(dispatch_get_main_queue(), ^{
-                    completionBlock();
+                    completionBlock(writeError);
                 });
             }
         });
     } else {
         if (completionBlock) {
-            completionBlock();
+            completionBlock(nil);
         }
     }
 }
 
-- (void)storeImageDataToDisk:(nullable NSData *)imageData forKey:(nullable NSString *)key {
+- (BOOL)storeImageDataToDisk:(nullable NSData *)imageData
+                      forKey:(nullable NSString *)key
+                       error:(NSError * _Nullable * _Nullable)errorPtr {
     if (!imageData || !key) {
-        return;
+        return YES;
```

Comment:
I think we should return `NO` here since the store operation did not go successfully.